deploy/terraform/main.tf

```diff
@@ -10,43 +10,46 @@ provider "packet" {
 }
 
 # Create a new VLAN in datacenter "ewr1"
-resource "packet_vlan" "provisioning-vlan" {
-  description = "provisioning-vlan"
+resource "packet_vlan" "provisioning_vlan" {
+  description = "provisioning_vlan"
   facility    = var.facility
   project_id  = var.project_id
 }
 
 # Create a device and add it to tf_project_1
-resource "packet_device" "tink-provisioner" {
+resource "packet_device" "tink_provisioner" {
   hostname         = "tink-provisioner"
   plan             = var.device_type
   facilities       = [var.facility]
   operating_system = "ubuntu_18_04"
   billing_cycle    = "hourly"
   project_id       = var.project_id
   user_data        = file("install_package.sh")
+}
+
+resource "null_resource" "tink_directory" {
```

Comment:
@gianarb by moving the provisioning to a `null_resource`, provisioning failures (like a missing ssh key) will not require the `tink_provisioner` host to be recreated.  Changes to the tink/ directory can be reapplied (rsync'd) independently. `terraform taint null_resource.tink_directory; terraform apply`